pints/_optimisers/_irpropmin.py

```diff
@@ -89,6 +109,15 @@ def __init__(self, x0, sigma0=0.1, boundaries=None):
         # Current step sizes
         self._step_size = np.array(self._sigma0)
 
+        # Rectangular boundaries
+        self._lower = self._upper = None
+        if isinstance(self._boundaries, pints.RectangularBoundaries):
+            self._lower = self._boundaries.lower()
+            self._upper = self._boundaries.upper()
```

Comment:
I am wondering whether it's more clear to move this to line 244. The self._lower and self._upper are not used in any other methods are they? And to me, it's not immediately clear that line 244 is only being invoked for rectangular boundaries.

Reply:
🤔 
I want to limit the number of function calls & inspection though, so no `isinstance` or `lower()` in `tell()`.
Not a deal-breaker though. Happy to use the isinstance check inside tell if you prefer

(I'm being efficient here out of habit, not really out of design)

Reply:
I am happy either way 😊 The way it is right now is probably easy enough to understand, but I would be surprised if the boundary method calls have a significant impact on the run time.